File: ios/Classes/data/PluginImage.swift
//
//  PluginImage.swift
//  zpdl_studio_media_plugin
//
//  Created by 김경환 on 2020/10/13.
//

import Foundation

struct PluginImage: PluginToMap {
    let id: String
    let width: Int
    let height: Int
    let modifyTimeMs: TimeInterval

    func pluginToMap() -> [String : Any] {
        return [
            "id": id,
            "width": width,
            "height": height,
            "modifyTimeMs": modifyTimeMs
        ]
    }
}

File: android/src/main/kotlin/com/zpdl_studio/zpdl_studio_media_plugin/data/PluginImage.kt
package com.zpdl_studio.zpdl_studio_media_plugin.data

data class PluginImage(
        val id: Long,
        val fullPath: String,
        val displayName: String,
        val mimeType: String,
        val width: Int,
        val height: Int,
        val modifyTimeMs: Long
) : PluginToMap {
    override fun pluginToMap(): Map<String, *> = hashMapOf(
            "id" to id,
            "width" to width,
            "height" to height,
            "modifyTimeMs" to modifyTimeMs,
            "info" to hashMapOf(
                    "fullPath" to fullPath,
                    "displayName" to displayName,
                    "mimeType" to mimeType
            )
    )
}